torch/_refs/__init__.py

```diff
@@ -941,6 +958,137 @@ def permute(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
     return prims.transpose(a, _permutation)
 
 
+def _reshape_view_helper(
+    a: TensorLikeType, shape: ShapeType, *, allow_copy: bool
+) -> TensorLikeType:
+    # NOTE: Reshape may be given a shape with a -1 length
+    # This indicates that the dimension's length should be inferred
+    # Creates a valid shape
+    for idx in range(len(shape)):
+        if shape[idx] == -1:
+            # Verifies there's only one dimension of length -1 in the shape
+            if shape.count(-1) > 1:
+                msg = "Can only infer the length of one dimension, but got shape {0}!".format(
+                    str(shape)
+                )
+                raise ValueError(msg)
+
+            length = reduce(operator.floordiv, (x for x in shape if x != -1), a.numel())
```

Comment:
This will catch the error of trying to reshape 0-element tensor to (0, -1) shape (which is good), but the error message will be confusing:
```
----> 1 length = reduce(operator.floordiv, (x for x in shape if x != -1), numel)

ZeroDivisionError: integer division or modulo by zero
```

Reply:
That's a good point -- added a comment to improve the error message